### PR TITLE
rac-282 integrate larval stream-monitors (and log handles)

### PR DESCRIPTION
**Pull request by stuart-stanley**
Note: don't actually merge until new output is cleared as "good enough" to go live. HOWEVER, there is a lot of code here, so getting the bulk of the review done would be VERY wise. Just remember this is steps on a path. There are todo's and holes in this.

Change to common/fit_common.py
* bring in new "flogging" (fit-logging) module from stream-monitor plugin space
* adjust run_nose() method to:
** fix: not use lucky-scoping to get access to noseopts variable
** converted invocation to use the nose runner instead of building a monster shell command.
*** env passed in via official api
*** noseargs passed in the same
*** converted from dangerous string to list-of-args
*** interpret success/failure out of nose.run as 0/1 to allow test_runner.py to sys.exit() via it.
* added run_from_module() method (see changes to tests/rackhd11/test_rackhd11_api_catalogs.py)

Everything under stream-monitor is basically the plugin from PR https://github.com/stuart-stanley/nose-observer-plugin/pull/1.
It contains four sub-dirs and some support files. This is implemented as a plugin for nose!
* support files:
** readme, mkenv.sh, and requirements.
* sm_plugin:
** the real plugin base. basically just enough smarts right now to orchestrate telling future stream-sources "test starting!" "test ending!" and so on.
* stream-sources: where all the actual stream "sources" will live. At the moment, this contains a (test harness) logging stream monitor:
** smart enough to detect our local-loggers
** able to inject additional log message to bracket test start/stops
** ...which also means those contact points will be available to use a common scanner (did I see XXX? Yeah? GO BOOM!) and so on.
* flogging is the code for:
** rac-263 (basic log handles for use in test areas)
** rac-264 (granular logging levels)
** this code is meant to be used from fit_tests via common/fit_common.py:get_loggers()

tests/rackhd11/test_rackhd11_api_catalogs.py shows an example conversion:
* bring in the multi-loggers holder
* convert a couple of prints to use them:
** mlgs.trl -> test run logger
** mlgs.tdl -> test data logger
** demo the use of debug_7 (logger levels) vs just "debug"
* convert __name__ module invoker to use new run_from_module to nose this specific file

Stragglers:
* touched up requirements file
* Update Vagrant to use correct mkenv steps
* Update README to use correct mkenv steps.

### CODE REVIEW

File: test/fit_tests/tests/rackhd11/test_rackhd11_api_catalogs.py [hunk 1]
```diff
@@ -25,22 +26,25 @@ def test_api_11_catalogs(self):
         self.assertEqual(api_data['status'], 200,
                         'Incorrect HTTP return code, expected 200, got:' + str(api_data['status']))
         self.assertNotEqual(len(api_data['json']), 0, "Error, no catalog")
+        logs.debug('examing %d entries from catalogs', len(api_data['json']))
```
Comment (johren) [inline test/fit_tests/tests/rackhd11/test_rackhd11_api_catalogs.py]: Typo - examining

File: test/fit_tests/requirements.txt [hunk 2]
```diff
@@ -1,4 +1,11 @@
-requests==2.2.1
+gevent==1.1.2
+greenlet==0.4.10
+nose==1.3.7
 pexpect==3.1
 pika==0.10.0
-nose==1.3.7
+pkg-resources==0.0.0
```
Comment (hohene) [inline test/fit_tests/requirements.txt]: Hitting an error on the pkg-resources requirement when setting up my virtual test env.
```
Downloading/unpacking pika==0.10.0 (from -r requirements.txt (line 5))
  Downloading pika-0.10.0-py2.py3-none-any.whl (92kB): 92kB downloaded
Downloading/unpacking pkg-resources==0.0.0 (from -r requirements.txt (line 6))
  Could not find any downloads that satisfy the requirement pkg-resources==0.0.0 (from -r requirements.txt (line 6))
Cleaning up...
No distributions at all found for pkg-resources==0.0.0 (from -r requirements.txt (line 6))
Storing debug log for failure in /emc/hohene/.pip/pip.log
```

File: test/fit_tests/common/fit_common.py [hunk 3]
```diff
@@ -859,47 +865,61 @@ def apply_obm_settings_seq():
...
+            'HTTPS':  str(ARGS_LIST['https']),
+            'PORT':  str(ARGS_LIST['port'])
+        }
+        argv = ['nosetests']
+        argv.extend(noseopts)
+        argv.append('--xunit-file')
+        argv.append(xmlfile)
+        argv.extend(pathspec)
+        if nose.run(argv=argv, env=env):
+            return 0
```
Comment (hohene) [inline test/fit_tests/common/fit_common.py]: This is still biting me running the tests, when the subprocess command is removed, it doesn't work.
the variables set via args_list aren't being used...i get "localhost" for the ora.

File: test/fit_tests/stream-monitor/stream_sources/log_type.py [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+            fn, lno, func = "(unknown file)", 0, "(unknown function)"
+        if exc_info:
+            if not isinstance(exc_info, tuple):
+                exc_info = sys.exc_info()
+        record = logger.makeRecord(logger.name, INFO, fn, lno, msg, args, exc_info, func, extra)
+        for handler in handlers:
+            handler.handle(record)
+
+    def __mark_all_loggers(self, bracket, fmat, *args, **kwargs):
+        rep_cnt = int(70/len(bracket))
```
Comment (hohene) [inline test/fit_tests/stream-monitor/stream_sources/log_type.py]: I'm finding in a full log, the rep_cnt of 70 make it hard to pick out the test name, can you drop it to 8 so it stands out a bit more. 
```
2016-12-08 16:13:16,327 547 MainProcess /emc/hohene/sandbox-github/rackhd/stu-log/test/fit_tests/stream-monitor/stream_sources/log_type.py:handle_start_test@29 greenlet:main INFO +0+0+0+0+0+0+0+0+0+0+0+0+0+0+0+0+0+0+0+0+0+0+0+0+0+0+0+0+0+\
0+0+0+0+0+0
2016-12-08 16:13:16,327 547 MainProcess /emc/hohene/sandbox-github/rackhd/stu-log/test/fit_tests/stream-monitor/stream_sources/log_type.py:handle_start_test@29 greenlet:main INFO -------STARTING TEST: test_api_11_catalogs (rackhd11.test_r\
ackhd11_api_catalogs.rackhd11_api_catalogs)--------
2016-12-08 16:13:16,327 547 MainProcess /emc/hohene/sandbox-github/rackhd/stu-log/test/fit_tests/stream-monitor/stream_sources/log_type.py:handle_start_test@29 greenlet:main INFO +0+0+0+0+0+0+0+0+0+0+0+0+0+0+0+0+0+0+0+0+0+0+0+0+0+0+0+0+0+\
0+0+0+0+0+0
```
```
2016-12-08 16:13:16,327 547 MainProcess /emc/hohene/sandbox-github/rackhd/stu-log/test/fit_tests/stream-monitor/stream_sources/log_type.py:handle_start_test@29 greenlet:main INFO +0+0+0+0
2016-12-08 16:13:16,327 547 MainProcess /emc/hohene/sandbox-github/rackhd/stu-log/test/fit_tests/stream-monitor/stream_sources/log_type.py:handle_start_test@29 greenlet:main INFO -------STARTING TEST: test_api_11_catalogs (rackhd11.test_r\
ackhd11_api_catalogs.rackhd11_api_catalogs)--------
2016-12-08 16:13:16,327 547 MainProcess /emc/hohene/sandbox-github/rackhd/stu-log/test/fit_tests/stream-monitor/stream_sources/log_type.py:handle_start_test@29 greenlet:main INFO +0+0+0+0
```

File: test/fit_tests/requirements.txt (hunk 2 above)
Comment (hohene) [inline test/fit_tests/requirements.txt]: this error was due to running from ubuntu 14.04 vs 16.

File: test/fit_tests/stream-monitor/stream_sources/log_type.py (hunk 4 above)
Comment (stuart-stanley) [inline test/fit_tests/stream-monitor/stream_sources/log_type.py]: changed to fill 20 characters instead of 70.

**Comment by stuart-stanley** [comment]
asked to move to integration branch...

**Pull request closed without merging**